test/functional/admin/link_check_reports_controller_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'test_helper'
+require "gds_api/test_helpers/link_checker_api"
+
+class Admin::LinkCheckReportsControllerTest < ActionController::TestCase
+  include GdsApi::TestHelpers::LinkCheckerApi
+
+  setup do
+    login_as :writer
+    @publication = create(:publication, body: "[link](http://www.example.com)")
+    Rails.backtrace_cleaner.remove_silencers!
+
+    link_checker_endpoint = "#{Plek.find('link-checker-api')}/batch"
+    body = link_checker_api_batch_report_hash(
+      id: 5,
+      links: [{ uri: "http://www.example.com" }],
+    )
+    stub_request(:post, %r{\A#{link_checker_endpoint}})
```

Comment:
There are some helper methods which can stub the whole request, have you tried those?